### PR TITLE
First batch of code smells

### PR DESCRIPTION
**Pull request by siy**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/integration/java/com/radixdlt/harness/deterministic/DeterministicActorsTest.java [hunk 1]
```diff
@@ -149,19 +149,16 @@ protected void setActorConfigurations(List<ActorConfiguration> actorConfiguratio
...
-            nodeKeys.stream()
-                .map(k -> BFTNode.create(k.getPublicKey()))
-                .collect(Collectors.toList()),
+            nodeKeys.stream().map(k -> BFTNode.create(k.getPublicKey())).toList(),
             MessageSelector.firstSelector(),
             MessageMutator.nothing());
 
-    List<BFTNode> allNodes =
-        nodeKeys.stream().map(k -> BFTNode.create(k.getPublicKey())).collect(Collectors.toList());
+    List<BFTNode> allNodes = nodeKeys.stream().map(k -> BFTNode.create(k.getPublicKey())).toList();
```
Comment (diegomrsantos) [inline radixdlt-core/radixdlt/src/integration/java/com/radixdlt/harness/deterministic/DeterministicActorsTest.java:156]: is there a repetition between line 152 and 156?

Comment (siy) [reply radixdlt-core/radixdlt/src/integration/java/com/radixdlt/harness/deterministic/DeterministicActorsTest.java]: Indeed. Good catch.
Fixed.

File: radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java [hunk 2]
```diff
@@ -139,6 +139,10 @@ public int getId() {
     return id;
   }
 
+  public boolean sameAs(int id) {
```
Comment (diegomrsantos) [inline radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java:142]: Is it really necessary?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java [hunk 3]
```diff
@@ -86,13 +83,13 @@ protected void configure() {
 
   @Provides
   @Singleton
-  private Forks forks(TreeMap<Long, ForkConfig> forkConfigs) {
+  private Forks forks(NavigableMap<Long, ForkConfig> forkConfigs) {
```
Comment (diegomrsantos) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java:86]: why is it necessary?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java [hunk 4]
```diff
@@ -103,23 +102,17 @@
...
-      this.p = p;
-    }
-
+  private record Subscription<T>(Observable<T> o, EventProcessor<T> p) {
     Disposable subscribe(Scheduler s) {
       return o.observeOn(s)
           .subscribe(
               p::process,
               e -> {
                 // TODO: Implement better error handling especially against Byzantine nodes.
```
Comment (diegomrsantos) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java:111]: Can we remove the TODOs and add tasks in the backlog instead?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/bft/VerifiedVertexStoreState.java [hunk 5]
```diff
@@ -167,28 +166,30 @@ public static VerifiedVertexStoreState create(
...
       if (header.getView().gt(root.getView())) {
-        VerifiedVertex newRoot = idToVertex.get(header.getVertexId());
-        ImmutableList<VerifiedVertex> newVertices =
+        var newRoot = idToVertex.get(header.getVertexId());
+        var newVertices =
             ImmutableList.of(
                 idToVertex.get(highQC.highestQC().getParent().getVertexId()),
                 idToVertex.get(highQC.highestQC().getProposed().getVertexId()));
-        ImmutableMap<HashCode, VerifiedVertex> idToVertex =
+        var idToVertexMap =
```
Comment (diegomrsantos) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/bft/VerifiedVertexStoreState.java:185]: do we need to add "Map" to the name?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/EngineConfigurationHandler.java [hunk 6]
```diff
@@ -78,18 +78,18 @@
 import com.radixdlt.statecomputer.checkpoint.Genesis;
 import com.radixdlt.statecomputer.forks.ForkConfig;
 import com.radixdlt.utils.Bytes;
-import java.util.TreeMap;
+import java.util.NavigableMap;
 
 public final class EngineConfigurationHandler
     extends CoreJsonRpcHandler<EngineConfigurationRequest, EngineConfigurationResponse> {
-  private final TreeMap<Long, ForkConfig> forks;
+  private final NavigableMap<Long, ForkConfig> forks;
```
Comment (diegomrsantos) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/EngineConfigurationHandler.java:85]: why is it necessary?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/RadixNodeModule.java [hunk 7]
```diff
@@ -123,11 +125,10 @@
...
   public RadixNodeModule(RuntimeProperties properties) {
     this.properties = properties;
-    var networkId = properties.get("network.id");
-    if (networkId == null) {
-      throw new IllegalStateException("Must specify network.id");
-    }
-    this.networkId = Integer.parseInt(networkId);
+    this.networkId =
+        Optional.ofNullable(properties.get("network.id"))
+            .map(Integer::parseInt)
```
Comment (diegomrsantos) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/RadixNodeModule.java:130]: This is gonna throw a `NumberFormatException` if the id isn't a parsable integer.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/EngineConfigurationHandler.java (hunk 6 above)
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/EngineConfigurationHandler.java:85]: this map is going away with the coordinated forks branch

File: radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java (hunk 2 above)
Comment (siy) [reply radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java:142]: It enabled me to cleanup logic and reduce method complexity.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java (hunk 3 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java:86]: TreeMap is a concrete class, NavigableMap is an interface. Basically I've just fixed one of the code smells reported by Sonar.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java (hunk 4 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java:111]: - Adding task does not necessarily result to removing of TODO, as comment points exactly to the place where issue is present.
- Adding tickets for TODO's is a separate task, I just had no time for this.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/bft/VerifiedVertexStoreState.java (hunk 5 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/bft/VerifiedVertexStoreState.java:185]: The original name shadowed the other variable, and it was reported by Sonar as code smell.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/EngineConfigurationHandler.java (hunk 6 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/EngineConfigurationHandler.java:85]: See above, this change was made as part of cleanup of code smell.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/RadixNodeModule.java (hunk 7 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/RadixNodeModule.java:130]: Yes. Exactly like original code does.

File: radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java (hunk 2 above)
Comment (talekhinezh) [reply radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java:142]: Agree with @diegomrsantos . Creating another method to simplify and add another layer of indirection is not worth it:
`if (networkId == Network.MAINNET.getId()) {` to
`if (Network.MAINNET.sameAs(networkId)) {`

**Review by LukasGasior1 — APPROVED** [review]

File: radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java (hunk 2 above)
Comment (siy) [reply radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java]: The second version does not make the assumption that the reader knows about the meaning and properties of network ID (i.e. that all ID's must be distinct). Nevertheless, I see no point to argue and will roll back changes.

Comment (diegomrsantos) [reply radixdlt-java-common/src/main/java/com/radixdlt/networks/Network.java]: The notion of equality is already defined in the equals method.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java (hunk 3 above)
Comment (diegomrsantos) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java:86]: ah cool, I didn't remember that.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java (hunk 4 above)
Comment (diegomrsantos) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java:111]: Doesn't sonar consider TODO a code smell?

**Pull request merged**